### ISSUE TITLE
Text not correctly rendered in History diff

### ISSUE THREAD

**Issue by vczh** [desktop/desktop#3923]
### Description

When a source file contains some character that does not appear in the current locale, but the related language package is installed, those code are rendered correctly if you read the diff in Changes tab, but they are not rendered correctly if you read the diff in History tab.

### Version

**GitHub Desktop version:** 1.0.13

**OS version:** Microsoft Windows [Version 10.0.16299.192]

### Steps to Reproduce

1. Set your computer's locale to english
2. Install Chinese language package
3. Type some Chinese character to your code
4. You can see diff shows Chinese characters when you are reading diff from Changes tab
5. Commit it
6. Open History tab and read the diff, those Chinese characters are rendered as unrecognizable symbols

**Expected behavior:** Character correctly rendered (see repro steps for details)

**Actual behavior:** Character incorrectly rendered (see repro steps for details)

**Reproduces how often:** 100%

### Logs

This is not a crash so I don't think log helps.

#### Additional Information

You can use Github Desktop to read this change: https://github.com/vczh-libraries/GacUI/commit/92945be6a7fe72c037409cb60372e8ca5be3975f , remember to install Chinese language pack while keeping your computer in English. Undo this commit and see things just work in Changes tab, but not in History tab.

**Comment by iAmWillShepherd** [comment]
Thanks for taking the time to open an issue! 🙇 

I was unable to reproduce this problem on my computer (macOS).

![2018-02-05_13-35-57 197](https://user-images.githubusercontent.com/1715082/35824819-b3f04080-0a79-11e8-9bf5-67d21bb9c414.png)

![2018-02-05_13-36-04 838](https://user-images.githubusercontent.com/1715082/35824837-c0d72a20-0a79-11e8-817b-27c8a0d9782d.png)

**Comment by vczh** [comment]
@iAmWillShepherd I am using Windows 10 which is mentioned in the description.

**Comment by iAmWillShepherd** [comment]
Yes, I went ahead and tried to see if I could reproduce on macOS to rule it out. I'm having trouble with my Windows VM, but I intend to check this out there as well.

**Comment by iAmWillShepherd** [comment]
@vczh I was unable to reproduce on Windows 10 as well.

![2018-02-23 11 56 42](https://user-images.githubusercontent.com/1715082/36608873-c28238c2-1890-11e8-8456-808f6122161c.gif)

**Comment by shiftkey** [comment]
@iAmWillShepherd could there be something in the problem repository and commits like this that are different to your setups? https://github.com/vczh-libraries/GacUI/commit/92945be6a7fe72c037409cb60372e8ca5be3975f

**Comment by shiftkey** [comment]
Confirmed the problem commit https://github.com/vczh-libraries/GacUI/commit/92945be6a7fe72c037409cb60372e8ca5be3975f does not render correctly in Desktop for Windows _and_macOS, so I think we need to figure out a way to drop this constraint on rendering UTF-8 diffs.

**Comment by shiftkey** [comment]
@iAmWillShepherd yeah, this basically means encoding detection and/or overriding support not unlike #1607

**Comment by shiftkey** [comment]
@vczh this is something I looked into ages ago, but it didn't eventuate: https://github.com/desktop/desktop/pull/634

**Comment by GnarlyDave** [comment]
I'm not sure that it's just a problem with History.
I'm getting this on viewing uncommited changes on a UTF-8 file CSV (without BOM) with diacritics.
Is it simply Windows 10 being told to use a font that doesn't have enough glyphs?

**Comment by vczh** [comment]
@GnarlyDave Obviously not related to fonts. @shiftkey has repro-ed and confirmed it is an encoding issue.

**Comment by zerox1212** [comment]
I have the same issue in the Changes tab when working with ANSI (Windows 1252) encoded text files. Special ANSI characters render as "?" blocks.

**Comment by decasamerlo** [comment]
Do we have any solution on this? Having big trouble here :(

**Comment by shiftkey** [comment]
@songshinyoung opened #7671 to indicate that this problem is still noticeable, and I checked one of the sample commits from earlier https://github.com/vczh-libraries/GacUI/commit/92945be6a7fe72c037409cb60372e8ca5be3975f and confirmed this is still an issue on `1.7.1-beta1`:

<img width="1246" src="https://user-images.githubusercontent.com/359239/58800575-9fb23100-85de-11e9-9037-30c73134e05b.png">

**Comment by shiftkey** [comment]
@sk74575 opened #7733 which overlaps with this:

> when i was write to korean it doen't work i think my program problem app wasn't use utf-8 so help me

**Comment by shiftkey** [comment]
@gmikhail opened #7731 which overlaps with this:

> ## Steps to Reproduce
> 1. `Add the file with cyrillic text and Windows-1251 encoding`
> 2. `See the commit preview`

**Comment by shiftkey** [comment]
@ntsoftabc opened #7632 which is potentially in this same vein

> ## Steps to Reproduce
> 1. `Add some Chinese/Japanese characters for comments`
> 2. `Can Not display(Unreadable for humans)`

**Comment by vczh** [comment]
I am curious that, if characters are rendered correctly in uncommitted change diff, then somewhere should have been processing encoding correctly. Why this does not happen also in History tab?

**Comment by zerox1212** [comment]
I noticed this same thing in Atom text editor. It could be that the bug is actually in Electron.js framework.

**Comment by lesterw1** [comment]
See https://github.com/desktop/desktop/issues/7857 for the sample [test.txt](https://github.com/desktop/desktop/files/3321613/test.txt) file. This issue still exists in v2.2.4 of Github Desktop.

**Comment by vczh** [comment]
> See #7857 for the sample [test.txt](https://github.com/desktop/desktop/files/3321613/test.txt) file. This issue still exists in v2.2.4 of Github Desktop. Perhaps there is a way to force files with specific extensions to be treated as text (i.e., convert to UTF-8)? I encountered the same issue with a page.html file.

This is not safe

**Comment by jredfox** [comment]
add ability in the desktop gui to play with the `.gitconfig` and `.gitignore` these backend files almost no one knows how to use do to no good documentation and it not being in the desktop gui. the ignore kind of is. I literally look it up on google and it tells you to use a linux command which doesn't exist on windows and then it doesn't work for `.gitignore` nor do I know how to use it

**Comment by mlferreira151276** [comment]
I'm working with the latest version of Github Desktop and I still have problems with ANSI files

![image](https://user-images.githubusercontent.com/45496139/163189599-20758410-a959-4632-a0ed-2d661737798a.png)

I'm working with the latest version of Github Desktop and I still have problems with ANSI files

**Comment by arvanus** [comment]
Same problem here, any idea of when a fix will be available? I'm using `git-gui` when I commit chars like `áçãàü` 😢 

![image](https://user-images.githubusercontent.com/8182958/202500301-03a77316-12fd-4e14-bdef-2c3605b3610c.png)

**Comment by FozerG** [comment]
Cyrillic with Windows-1251 encoding is not displayed correctly.
![image](https://user-images.githubusercontent.com/78788357/204054832-5128c5f6-c2e1-4371-b2c9-8d5588a19a8c.png)

**Comment by steveward** [comment]
Another report in https://github.com/desktop/desktop/issues/17662.

**Comment by steveward** [comment]
From https://github.com/desktop/desktop/issues/18030:

> I have some problems with displaying characters in the GitHub Desktop. My file vers.info is in cp1251 encoding. And .gitattributes contains the next line "*.info working-tree-encoding=C
> P1251 eol=CRLF". In the new part of the file i see correct cyrillic characters, but in the old part I see question signs inside diamonds on the cyrillic characters places.

**Comment by murchelon** [comment]
> I'm not sure that it's just a problem with History. I'm getting this on viewing uncommited changes on a UTF-8 file CSV (without BOM) with diacritics. Is it simply Windows 10 being told to use a font that doesn't have enough glyphs?

I have the exact same situation: some characters are showing up as a question mark, in a CSV file, in ansi format. If I use a UTF8 file, there is no problem.

**Comment by steveward** [comment]
Reported in https://github.com/desktop/desktop/issues/19530.

**Comment by UnrealKaraulov** [comment]
Same issue

![Image](https://github.com/user-attachments/assets/56765e52-4292-4656-814f-5c82ca5e5a5f)

how to fix it?

**Comment by songshinyoung** [comment]
I am using Windows 10.  
The language is set to Korean.  
When my file is of the "Ansi" type, there is an issue where Korean comments break due to the same problem.  
I confirmed that after changing the problematic file to the UTF-8 type, the Korean comments no longer break.  
I recommend a tool or method to change file encoding to UTF-8.
However, support is needed to ensure that characters do not break, regardless of the type of file.

**Comment by vico93** [comment]
Is still an issue for me - i sometimes do things in PAWN for Open.MP (a spiritual sucessor of SA:MP) that requires the source files to be encoded as Windows-1252 (ANSI). Diffs of those files always show the accents with the "?" character.

<img width="572" height="68" alt="Image" src="https://github.com/user-attachments/assets/65c544e3-7bde-44df-a9a4-699dd23354c6" />

**Comment by wezzzyrek1** [comment]
Same here. If file is encoded using ANSI charset then all breaks

<img width="1860" height="977" alt="Image" src="https://github.com/user-attachments/assets/f18618ff-62c9-45c2-9c2f-4f802dee6139" />

In this case the file is encoded using 949 Korean. Changing gile encoding to UTF8 is not a solution in many cases, as we use 949 on purpose.
